code/game/atoms.dm

```diff
@@ -465,10 +465,11 @@ Proc for attack log creation, because really why not
 
 	if(light_power && light_range)
 		update_light()
-
-	if(opacity && isturf(loc))
-		var/turf/T = loc
-		T.has_opaque_atom = TRUE // No need to recalculate it in this case, it's guaranteed to be on afterwards anyways.
+	if(loc)
+		SEND_SIGNAL(loc, COMSIG_ATOM_INITIALIZED_ON, src) //required since spawning something doesn't call Move hence it doesn't call Entered.
```

Comment:
```suggestion
		SEND_SIGNAL(loc, COMSIG_ATOM_INITIALIZED_ON) //required since spawning something doesn't call Move hence it doesn't call Entered.
```
`src` is sent by default as the first argument.

Reply:
It is not. loc is sent by default as the first argument.

Reply:
Oh, I misread that, you are sending the signal to loc. Nevermind.